### PR TITLE
docs: expand jfrog platform and example template

### PR DESCRIPTION
**Pull request by matifali**
1. Updates to use `artifactory_scoped_token`
2. Updates the terraform provider
3. Adds examples for `Pypi` and `Go`.
4. Refactors the template to be usable with other package managers

### CODE REVIEW

**Review by ammario — APPROVED** [review]

File: docs/platforms/jfrog.md [hunk 1]
```diff
@@ -57,15 +57,15 @@ variable "artifactory_access_token" {
...
   access_token  = "${var.artifactory_access_token}"
 }
 ```
 
-When pushing the template, you can pass in the variables using the `-V` flag:
+When pushing the template, you can pass in the variables using the `--var` flag:
 
 ```sh
-coder templates push --var 'jfrog_url=https://YYY.jfrog.io' --var 'artifactory_access_token=XXX'
+coder templates push --var 'jfrog_url=YYY.jfrog.io' --var 'artifactory_access_token=XXX'
```
[the remark this answers is not in the document]
Comment (matifali) [reply docs/platforms/jfrog.md:68]: I will update.

**Review by ammario — APPROVED** [review]

File: examples/templates/jfrog-docker/main.tf [hunk 2]
```diff
@@ -10,13 +10,23 @@ terraform {
...
 locals {
-  username = data.coder_workspace.me.owner
+  # if the jfrog username is same as the coder username, you can use the following
+  # artifactory_username = data.coder_workspace.me.owner
+  # if the username is same as email, you can use the following
+  # artifactory_username = urlencode(data.coder_workspace.me.owner_email)
+  artifactory_username = data.coder_workspace.me.owner
+  artifactory_registry_keys = {
+    "npm"    = "npm"
+    "python" = "python"
```
Comment (yinonavraham) [inline examples/templates/jfrog-docker/main.tf:26]: In the example in the docs you used `pypi`, which is also the default name in Artifactory, and I think it is better to keep the two examples (docs and this full example) aligned.
```suggestion
    "pypi" = "pypi"
```

File: examples/templates/jfrog-docker/main.tf [hunk 3]
```diff
@@ -67,23 +74,35 @@ resource "coder_agent" "main" {
...
-    registry=${var.jfrog_url}/artifactory/api/npm/npm/
+    registry = https://${var.jfrog_host}/artifactory/api/npm/${local.artifactory_registry_keys["npm"]}
     EOF
     jf rt curl /api/npm/auth >> .npmrc
+
+    # Configure the `pip` to use the Artifactory "pypi" registry.
+    mkdir -p ~/.pip
+    cat << EOF > ~/.pip/pip.conf
+    [global]
+    index-url = https://${local.artifactory_username}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/pypi/${local.artifactory_registry_keys["python"]}/simple
```
Comment (yinonavraham) [inline examples/templates/jfrog-docker/main.tf:91]: Following the other comment on aligning the names in both examples -
```suggestion
    index-url = https://${local.artifactory_username}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/pypi/${local.artifactory_registry_keys["pypi"]}/simple
```

File: docs/platforms/jfrog.md [hunk 4]
```diff
@@ -107,9 +107,27 @@ resource "coder_agent" "main" {
...
     jf c rm 0 || true
-    echo ${artifactory_access_token.me.access_token} | \
-      jf c add --access-token-stdin --url ${var.jfrog_url} 0
+    echo ${artifactory_scoped_token.me.access_token} | \
+      jf c add --access-token-stdin --url https://${var.jfrog_host} 0
+
+    # Configure the `npm` CLI to use the Artifactory "npm" registry.
+    cat << EOF > ~/.npmrc
+    email = ${data.coder_workspace.me.owner_email}
+    registry = https://${var.jfrog_host}/artifactory/api/npm/${local.artifactory_registry_keys["npm"]}
```
Comment (yinonavraham) [inline docs/platforms/jfrog.md:116]: The general term in Artifactory is "repository" -
```suggestion
    registry = https://${var.jfrog_host}/artifactory/api/npm/${local.artifactory_repository_keys["npm"]}
```

File: docs/platforms/jfrog.md [hunk 5]
```diff
@@ -107,9 +107,27 @@ resource "coder_agent" "main" {
...
+    EOF
+    jf rt curl /api/npm/auth >> .npmrc
+
+    mkdir -p ~/.pip
+    cat << EOF > ~/.pip/pip.conf
+    [global]
+    index-url = https://${data.coder_workspace.me.owner}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/pypi/${local.artifactory_registry_keys["pypi"]}/simple
```
Comment (yinonavraham) [inline docs/platforms/jfrog.md:123]:
```suggestion
    index-url = https://${data.coder_workspace.me.owner}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/pypi/${local.artifactory_repository_keys["pypi"]}/simple
```

File: docs/platforms/jfrog.md [hunk 6]
```diff
@@ -107,9 +107,27 @@ resource "coder_agent" "main" {
...
+    EOF
+
   EOT
+  # Set GOPROXY to use the Artifactory "go" registry.
+  env = {
+    GOPROXY : "https://${data.coder_workspace.me.owner}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/go/${local.artifactory_registry_keys["go"]}"
```
Comment (yinonavraham) [inline docs/platforms/jfrog.md:129]:
```suggestion
    GOPROXY : "https://${data.coder_workspace.me.owner}:${artifactory_scoped_token.me.access_token}@${var.jfrog_host}/artifactory/api/go/${local.artifactory_repository_keys["go"]}"
```

**Review by yinonavraham — COMMENTED** [review]
LGTM.
Few suggestions, mainly around naming and terminology.

File: examples/templates/jfrog-docker/main.tf (hunk 2 above)
Comment (matifali) [reply examples/templates/jfrog-docker/main.tf:26]: Thanks for the useful commnets. `pypi` is the key for a remote repository while `python` is a virtual repository that inlcludes `pypi`. I will sunchronise docs and template to use same keys.

**Pull request merged**